sqlmesh/core/config/connection.py

```diff
@@ -1422,8 +1428,19 @@ def _engine_adapter(self) -> t.Type[EngineAdapter]:
     @property
     def _connection_factory(self) -> t.Callable:
         from clickhouse_connect.dbapi import connect  # type: ignore
+        from clickhouse_connect.driver import httputil  # type: ignore
+        from functools import partial
 
-        return connect
+        pool_manager_options: t.Dict[str, t.Any] = dict(
+            num_pools=self.concurrent_tasks,
```

Comment:
As I understand from the docs the number of pools is equal to number of hosts or a max of 10. I don't see how it would correlate with the concept of `concurrent_tasks`.

Reply:
Some of this is definitely ignorance on my part. Let me review the docs more closely but I'm fine setting any value here really.

Reply:
Definitely correct, I'll make an update